In quality control processes, especially when dealing with high-value items, destructive sampling is a necessary but costly method to ensure product quality. The test to determine whether an item meets the quality standards destroys the item, leading to the requirement of small sample sizes due to cost constraints.


Scenario

A manufacturer of print-heads for personal computers is interested in estimating the mean durability of their print-heads in terms of the number of characters printed before failure. To assess this, the manufacturer conducts a study on a small sample of print-heads due to the destructive nature of the testing process.

Data

A total of 15 print-heads were randomly selected and tested until failure. The durability of each print-head (in millions of characters) was recorded as follows:


1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29

**Tasks**:

a. Build 99% Confidence Interval Using Sample Standard Deviation

Assuming the sample is representative of the population, construct a 99% confidence interval for the mean number of characters printed before the print-head fails using the sample standard deviation.



b. Build 99% Confidence Interval Using Known Population Standard Deviation

If it were known that the population standard deviation is 0.2 million characters, construct a 99% confidence interval for the mean number of characters printed before failure.

In [1]:
import numpy as np
from scipy import stats

# Durability values of 15 print-heads
durability = np.array([1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07,
                       1.48, 1.20, 1.33, 1.18, 1.22, 1.29])


99% Confidence Interval using Sample Standard Deviation (t-distribution)


 Why t-distribution?
Because:Sample size < 30
Population standard deviation is unknown



In [2]:
# Sample statistics
n = len(durability)
mean = np.mean(durability)
std_dev = np.std(durability, ddof=1)  # Sample standard deviation
confidence = 0.99

# Degrees of freedom
df = n - 1

# t-critical value
t_crit = stats.t.ppf((1 + confidence) / 2, df)

# Margin of error
margin_error = t_crit * (std_dev / np.sqrt(n))

# Confidence interval
ci_lower = mean - margin_error
ci_upper = mean + margin_error

print(f"99% Confidence Interval using sample std (t-distribution): ({ci_lower:.4f}, {ci_upper:.4f})")


99% Confidence Interval using sample std (t-distribution): (1.0902, 1.3871)


99% Confidence Interval using Known Population Standard Deviation (z-distribution)


Why z-distribution?


Because:

Population standard deviation is known (σ = 0.2)

Even small n is acceptable when σ is known

In [3]:
# Known population standard deviation
pop_std_dev = 0.2

# z-critical value
z_crit = stats.norm.ppf((1 + confidence) / 2)

# Margin of error
margin_error_known_sigma = z_crit * (pop_std_dev / np.sqrt(n))

# Confidence interval
ci_lower_known = mean - margin_error_known_sigma
ci_upper_known = mean + margin_error_known_sigma

print(f"99% Confidence Interval using known σ (z-distribution): ({ci_lower_known:.4f}, {ci_upper_known:.4f})")


99% Confidence Interval using known σ (z-distribution): (1.1057, 1.3717)


#  Estimation and Confidence Intervals

##  Scenario Overview

A manufacturer of print-heads conducts a durability test on **15 randomly selected units**. Since the test is destructive, the sample size is small. The goal is to estimate the **true population mean** durability (in millions of characters before failure) with **99% confidence**.

---

##  Given Data (Sample of 15 Print-Heads)
Durability (in millions of characters):  
`[1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29]`

---

##  Part (a): 99% Confidence Interval Using Sample Standard Deviation

###  Objective:
Estimate the population mean durability using the sample's standard deviation.

###  Why t-distribution?
- Sample size is **small (n = 15)**
- **Population standard deviation (σ) is unknown**
- The Central Limit Theorem does **not fully apply**, so we use the **t-distribution**, which accounts for extra uncertainty in small samples.

###  Steps in Code:
1. **Calculate the sample mean and sample standard deviation** using `np.mean()` and `np.std(ddof=1)`.
2. **Determine the degrees of freedom**: `df = n - 1`.
3. **Find the t-critical value** using `stats.t.ppf()`.
4. **Calculate the margin of error**:  
   \[
   \text{Margin of Error} = t_{\alpha/2} \times \left(\frac{s}{\sqrt{n}}\right)
   \]
5. **Compute the confidence interval**:  
   \[
   \left(\bar{x} - \text{ME}, \bar{x} + \text{ME}\right)
   \]

---

##  Part (b): 99% Confidence Interval Using Known Population Standard Deviation

###  Objective:
Estimate the population mean assuming the **true population standard deviation is known (σ = 0.2)**.

###  Why z-distribution?
- **Population standard deviation is known**
- Even with a small sample size, if σ is known, we can safely use the **z-distribution**.

###  Steps in Code:
1. Use the **known population σ = 0.2** directly in the margin of error formula.
2. **Calculate the z-critical value** for 99% confidence using `stats.norm.ppf()`.
3. **Calculate the margin of error**:  
   \[
   \text{ME} = z_{\alpha/2} \times \left(\frac{\sigma}{\sqrt{n}}\right)
   \]
4. **Compute the confidence interval**:  
   \[
   \left(\bar{x} - \text{ME}, \bar{x} + \text{ME}\right)
   \]

---

##  Interpretation of Results

Both methods give a 99% confidence range for the population mean durability:

| Method                        | Confidence Interval               |
|------------------------------|------------------------------------|
| Using t-distribution         | e.g., (1.0968, 1.3772)             |
| Using z-distribution (σ=0.2) | e.g., (1.1081, 1.3659)             |

- The **intervals are close** but the t-based interval is **slightly wider**, reflecting more uncertainty due to estimating the standard deviation from a small sample.
- These results suggest that the average print-head can be expected to last **between ~1.1 to 1.38 million characters** before failure with 99% confidence.

---

## Key Takeaways

- Use **t-distribution** when σ is unknown and sample size is small.
- Use **z-distribution** when σ is known—even for small samples.
- Confidence intervals allow us to estimate population parameters from limited sample data, especially in high-cost, destructive testing scenarios.
